app/services/duplicate_application.rb

```diff
@@ -58,6 +58,10 @@ def duplicate
       end
     end
 
+    original_application_form.application_references.feedback_requested.each do |reference|
+      reference.update_column(:hashed_sign_in_token, nil)
+    end
```

Comment:
Do we have to worry about tokens stored in `ReferenceToken#hashed_token`? If we still use `ReferenceToken` it seems we need to move (copy and delete) all the tokens from the old reference to the new one.

Reply:
Going off @davidgisbey guidance on this one, from what I understand the sign in token being set to nil prevents these old references from possibly being confused with the new ones.

There's probably an argument for deleting the references on the original application form unless there's any relevant auditng associated with them... David what are your thoughts?

Reply:
Wouldn't we have to copy the `ReferenceToken`s for the referees to be able to give feedback to the copied references?

Reply:
Yea good point. I missed that. so it would have to be something like

```
    original_application_form.application_references.where(feedback_status: %w[feedback_provided not_requested_yet cancelled_at_end_of_cycle feedback_requested]).reject(&:feedback_overdue?).each do |w|
      new_ref = new_application_form.application_references.create!(
        w.attributes.except(*IGNORED_CHILD_ATTRIBUTES).merge!(duplicate: true),
      )
      w.reference_token.update!(reference: new_ref)
      w.update_column(:hashed_sign_in_token, nil)
    end

```

Reply:
does that seem about right @stevehook ?

Reply:
Either way we should definitely add a system spec as this would have slipped through if it was just me reviewing!

Reply:
In the above code snippet `ApplicationReference#reference_token` is a single-value association, but I think it's a collection. So you'd have to loop over `ApplicationReference#reference_tokens` and do this to each one.

Reply:
Thanks both will have a go at rewriting and addinga spec

Reply:
So I decided in the end to reassign the reference token to the new duplicate reference rather than copying and deleting. Hope this approach makes sense please take a look and let me know. 

In terms of the system spec, I have already added `feedback_requested` references into each of the carry over system specs and these are all passing. However, when looking into each of these specs I can see that `ReferenceToken` is not assigned in any of these. Therefore i have opted to add a unit pec into the carry_over_application_spec to cover this functionality. Please let me know if this should go further.

Reply:
So i was thinking that the system spec would carry over an application, then have the referee respond and ensure the correct application reference was updated. Happy to add it if you're busy with support.

Reply:
Will have a crack, managing to balance things atm. Will let you know if I get swamped.